lib/beaker-pe/install/pe_utils.rb

```diff
@@ -1076,7 +1076,11 @@ def install_puppet_agent_pe_promoted_repo_on(hosts, opts)
             pe_ver = host[:pe_ver] || opts[:pe_ver] || '4.0.0-rc1'
             opts = sanitize_opts(opts)
             opts[:download_url] =
-              "#{opts[:pe_promoted_builds_url]}/puppet-agent/#{pe_ver}/#{opts[:puppet_agent_version]}/repos"
+              if pe_ver > '2023.7.0'
```

Comment:
relying on string compare is probably not the best idea here. I think we sould use the Semver module in beaker for this. https://github.com/voxpupuli/beaker/blob/657cb336bc1ae36c74fd95a230fd82dc90377eb2/lib/beaker/shared/semvar.rb#L14

Reply:
Revised the version comparison according to the suggestion.